bftengine/src/bcstatetransfer/Messages.hpp

```diff
@@ -66,11 +66,19 @@ struct CheckpointSummaryMsg : public BCStateTranBaseMsg {
   uint64_t requestMsgSeqNum;
 
   static bool equivalent(const CheckpointSummaryMsg* a, const CheckpointSummaryMsg* b) {
-    return (memcmp(a, b, sizeof(CheckpointSummaryMsg)) == 0);
+    static_assert((sizeof(CheckpointSummaryMsg) - sizeof(requestMsgSeqNum) == 82),
+                  "Should newly added field be compared below?");
+    return ((a->lastBlock == b->lastBlock) and (a->checkpointNum == b->checkpointNum) and
+            (memcmp(&a->digestOfLastBlock, &b->digestOfLastBlock, sizeof(STDigest)) == 0) and
+            (memcmp(&a->digestOfResPagesDescriptor, &b->digestOfResPagesDescriptor, sizeof(STDigest)) == 0));
   }
```

Comment:
STDigest has operator overloaded == :
`a->digestOfLastBlock == b->digestOfLastBlock`
Please change for 4 comparisions here and below.

Reply:
Thanks.